src/main/java/seedu/address/model/module/ModuleCode.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Module's module code in the GradPad.
+ * Guarantees: immutable; is valid as declared in {@link #isValidModuleCode(String)}
+ */
+public class ModuleCode {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Module codes should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+    A module code must have 2 or more characters as its prefix followed by 1 or more digits as its numerical
+    code. It can optionally end with 1 or more characters as a suffix. It is also case-insensitive.
+    */
+    public static final String VALIDATION_REGEX = "[\\p{Alpha}]{2,}\\d+[\\p{Alpha}]*";
```

Comment:
Let me know if this regex is fine cos I'm not too sure what kind of module codes we support

Reply:
I think should be fine since all the CS modules we are covering follows that format. Or should we limit to 7 characters?

Reply:
I think all NUS mods follow that format right? Unless we branch out to cover other Unis